config/config.php

```diff
@@ -1,33 +1,33 @@
 <?php
 
-use Zend\Stdlib\ArrayUtils;
-use Zend\Stdlib\Glob;
+use Zend\ConfigAggregator\ConfigAggregator;
+use Zend\ConfigAggregator\PhpFileProvider;
 
-/**
- * Configuration files are loaded in a specific order. First ``global.php``, then ``*.global.php``.
- * then ``local.php`` and finally ``*.local.php``. This way local settings overwrite global settings.
- *
- * The configuration can be cached. This can be done by setting ``config_cache_enabled`` to ``true``.
- *
- * Obviously, if you use closures in your config you can't cache it.
- */
+// To enable or disable caching, set the `ConfigAggregator::ENABLE_CACHE` boolean in
+// `config/autoload/local.php`.
+$cacheConfig = [
+    'config_cache_path' => 'data/config-cache.php',
+];
 
-$cachedConfigFile = 'data/cache/app_config.php';
+$aggregator = new ConfigAggregator([
+    //Include cache config for zf-development-mode
+    function () use ($cacheConfig) {
+        return $cacheConfig;
+    },
```

Comment:
The reason only callables or classes evaluating to callables are allowed is to make disambiguation easier, since an array _could_ also be callable. Secondarily, it's a performance optimization, as you can specify class names throughout, and thus have zero additional overhead in the case of a cache hit.

That said, we could have logic like this:

```php
if (! is_callable($provider) && is_string($provider) && class_exists($provider)) {
    $provider = new $provider();
}
if (! is_callable($provider) && is_array($provider)) {
    return $provider;
}
if (! is_callable($provider)) {
    throw new InvalidProviderException();
}
$config = $provider();
if (! is_array($config)) {
    throw new InvalidConfigFromProviderException();
}
return $config;
```

(pseudo-code; I'm not looking at the actual logic right now)

It adds some complexity, and also means there's more possible types allowable in the value, but, as you note, also means we don't have to perform any weird hacks to make it happen.

Would you be willing to open an issue and/or pull request for this against zend-config-aggregator, @MichaelGooden ?